src/libcommonserver/log/log.cpp

```diff
@@ -109,4 +115,393 @@ Log::Log(const log4cplus::tstring &filePath) {
+        LOG_WARN(Log::instance()->getLogger(),
+                 "Error in IoHelper::copyFileOrDirectory: " << Utility::formatIoError(parmsDbPath, ioError).c_str());
+
+        if (ioError == IoErrorDiskFull) {
+            exitCause = ExitCauseNotEnoughDiskSpace;
+        } else {
+            exitCause = ExitCauseUnknown;
+        }
+        return ExitCodeSystemError;
+    }
+    exitCause = ExitCauseUnknown;
+    return ExitCodeOk;
+}
+
+ExitCode Log::copyLogsTo(const SyncPath &outputPath, bool includeOldLogs, ExitCause &exitCause) {
+    SyncPath logPath = _filePath.parent_path();
+    IoError ioError = IoErrorUnknown;
+    IoHelper::DirectoryIterator dir;
+
+    if (!IoHelper::getDirectoryIterator(logPath, false, ioError, dir)) {
+        LOG_WARN(Log::instance()->getLogger(),
+                 "Error in DirectoryIterator: " << Utility::formatIoError(logPath, ioError).c_str());
+        exitCause = ExitCauseUnknown;
+        return ExitCodeSystemError;
+    }
+
+    DirectoryEntry entry;
+    bool endOfDirectory = false;
+    while (dir.next(entry, endOfDirectory, ioError) && !endOfDirectory) {
+        if (entry.is_directory()) {
+            LOG_WARN(Log::instance()->getLogger(), "Ignoring temp directory " << entry.path().filename().string().c_str());
+            continue;
+        }
+
+        if (!includeOldLogs && entry.path().filename().extension() == L".gz") {
+            LOG_WARN(Log::instance()->getLogger(), "Ignoring old log file " << entry.path().filename().string().c_str());
+            continue;
+        }
+
+        if (!IoHelper::copyFileOrDirectory(entry.path(), outputPath / entry.path().filename(), ioError)) {
+            LOG_WARN(Log::instance()->getLogger(),
+                     "Error in IoHelper::copyFileOrDirectory: " << Utility::formatIoError(entry.path(), ioError).c_str());
+            if (ioError == IoErrorDiskFull) {
+                exitCause = ExitCauseNotEnoughDiskSpace;
+                return ExitCodeSystemError;
+            } else {
+                exitCause = ExitCauseUnknown;
+                return ExitCodeSystemError;
+            }
+        }
+    }
+
+    if (!endOfDirectory) {
+        LOG_WARN(Log::instance()->getLogger(),
+                 "Error in DirectoryIterator: " << Utility::formatIoError(logPath, ioError).c_str());
+        exitCause = ExitCauseUnknown;
+        return ExitCodeSystemError;
+    }
+
+    exitCause = ExitCauseUnknown;
+    return ExitCodeOk;
+}
+
+ExitCode Log::compressLogFiles(const SyncPath &directoryToCompress, ExitCause &exitCause,
+                               std::function<void(int)> progressCallback) {
+    IoHelper::DirectoryIterator dir;
+    IoError ioError = IoErrorUnknown;
+    if (!IoHelper::getDirectoryIterator(directoryToCompress, true, ioError, dir)) {
+        LOG_WARN(Log::instance()->getLogger(),
+                 "Error in DirectoryIterator: " << Utility::formatIoError(directoryToCompress, ioError).c_str());
+        exitCause = ExitCauseUnknown;
```

Comment:
Same remark as before: you can move this statement above and remove the subsequent assignments below.